src/Support/Explain.php

```diff
@@ -21,49 +18,51 @@ public function confirm(string $connection): ?string
         };
     }
 
-    public function pack(string $connection, string $sql, ?array $bindings, int $validUntilTimestamp): ?string
+    public function hash(string $connection, string $sql, array $bindings): string
     {
+        $bindings = json_encode($bindings);
+
         return match (DB::connection($connection)->getDriverName()) {
-            'mysql', 'pgsql' => Crypt::encrypt(compact('connection', 'sql', 'bindings', 'validUntilTimestamp')),
+            'mysql', 'pgsql' => hash_hmac('sha256', "{$connection}::{$sql}::{$bindings}", config('app.key')),
             default => null,
         };
     }
 
-    private function unpack(string $encrypted): array
+    private function verify(string $connection, string $sql, array $bindings, string $hash): void
     {
-        try {
-            $data = Crypt::decrypt($encrypted);
-        } catch (DecryptException $e) {
-            throw new Exception('Query to execute could not be verified.', previous: $e);
-        }
+        $bindings = json_encode($bindings);
 
-        if ($data['validUntilTimestamp'] < time()) {
-            throw new Exception('Allowed time to analyze query has expired. Please reload the page.');
+        foreach ([config('app.key'), ...(config('app.previous_keys') ?? [])] as $key) {
```

Comment:
I don't think we need to take the previous keys into account here.

Reply:
Ok. Its now only using the current one.